src/Microsoft.Identity.Client/Instance/Discovery/IStaticMetadataProvider.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Microsoft.Identity.Client.Instance.Discovery
+{
+    internal interface IStaticMetadataProvider
+    {
+        void AddMetadata(string environment, InstanceDiscoveryMetadataEntry entry);
+        InstanceDiscoveryMetadataEntry GetMetadata(string environment);
+        void /* for test purposes */ Clear();
```

Comment:
This method isn't just for test.  It's called in instanceDiscoveryManager.cs line 39

Reply:
Yes, but that code path is for test purposes only (i.e. shouldClearCaches = true). I will add a comment InstanceDiscoveryManager to reflect that